lib/spack/llnl/util/lock.py

```diff
@@ -241,16 +309,21 @@ def acquire_read(self, timeout=None):
         """
         timeout = timeout or self.default_timeout
 
+        locktype = 'READ LOCK'
         if self._reads == 0 and self._writes == 0:
-            self._debug(
-                'READ LOCK: {0.path}[{0._start}:{0._length}] [Acquiring]'
-                .format(self))
+            self._log_acquiring(locktype)
             # can raise LockError.
             wait_time, nattempts = self._lock(fcntl.LOCK_SH, timeout=timeout)
-            self._acquired_debug('READ LOCK', wait_time, nattempts)
             self._reads += 1
+            self._log_acquired(locktype, wait_time, nattempts)
             return True
         else:
+            # Nothing is done with the provided lock timeout so log a warning
+            # if it is different from the one being used.
+            self._log_timeout_warning(timeout)
```

Comment:
why do we log this?  Do you use this for debugging somewhere?  I'm not sure why the user needs to know or what they'd do with this information.

Reply:
Excellent point.  Yes, it was intended as a debug message that I found useful.

Reply:
FWIW I was having trouble figuring out why some of the locks weren't behaving as I expected wrt timeouts.  These messages were helpful in tracking that issue down so I thought they would be worth keeping.

Reply:
Removed